src/vivarium_gates_nutrition_optimization_child/components/wasting.py

```diff
@@ -78,25 +76,28 @@ def get_current_exposure(self, index: pd.Index) -> pd.Series:
 
 class StaticChildWasting(Risk):
     def __init__(self):
-        """
-        Parameters
-        ----------
-        risk :
-            the type and name of a risk, specified as "type.name". Type is singular.
-        """
-        self.risk = EntityString("risk_factor.child_wasting")
-        name = "static_child_wasting"
-        self.configuration_defaults = self._get_configuration_defaults()
-        self.exposure_distribution = self._get_exposure_distribution()
-        self._sub_components = [self.exposure_distribution]
+        # use super's init to get exposure distribution
+        # but overwrite other names
+        risk = "risk_factor.child_wasting"
+        super().__init__(risk)
 
+        name = "static_child_wasting"
         self._randomness_stream_name = f"initial_{name}_propensity"
         self.propensity_column_name = f"{name}_propensity"
         self.propensity_pipeline_name = f"{name}.propensity"
         self.exposure_pipeline_name = f"{name}.exposure"
 
 
 class WastingTreatment(Risk):
+    @property
+    def time_step_prepare_priority(self) -> int:
+        # we want to reset propensities before updating previous state column
+        return 4
+
+    @property
+    def name(self) -> str:
+        return f"wasting_treatment_{self.risk}"
```

Comment:
Is there a reason you're manually defining this component's name?

Reply:
The name gets defined as wasting_treatment (no mention of self.risk) if I don't do this.

Reply:
The reason that happens is that your argument name is `treatment_type`, but it is not getting saved on the object as `treatment_type`, but rather `self.risk`. This is fine, but by convention we'd expect the name to be `f"wasting_treatment.{self.risk}"` with a `.` rather than a `_`.